mne/utils/numerics.py

```diff
@@ -748,7 +748,7 @@ def object_diff(a, b, pre=''):
         if b is not None:
             out += pre + ' left is None, right is not (%s)\n' % (b)
     elif isinstance(a, np.ndarray):
-        if not np.array_equal(a, b):
+        if not np.array_equal(a[~np.isnan(a)], b[~np.isnan(b)]):
```

Comment:
I would not do this silently. I would add a param to object_diff `check_nan` so you don't change the behavior for the other use cases